simpleviewinc/re-theme
# Zen 309/size keys fix

**Ticket**: [ZEN-309](https://jira.simpleviewtools.com/browse/ZEN-309)

> This PR includes Lance's changes outlined here: https://github.com/mpcarolin/re-theme/pull/2

## Context

* size-dynamic themes were not working in keg-components
* it seemed to be working in re-theme
* in keg-components, it only worked on the root object
* also, size keys did not have the `$` prefix, making it less clear what is a size-key and what is a style object name

## Goal

* fix the bug(s) to ensure that themes dynamically update
* update the size keys to include the `$`

## Updates

* `app/src/components/useTheme/useThemeViewportExample.js` and `app/src/screens/app.js`
  * implemented a simple example of size-dynamic themes, used in the re-theme `app/`
* `app/src/theme/components.js`
  * added the example theme with size keys
  * this justifies the major version update, since preexisting themes using default size keys without the `$` will stop working
* `app/src/theme/index.js`
  * removed the call to setDefaultTheme, since theme gets passed to `buildTheme` anyway and is built there
  * this also fixes a bug in the app that prevented the theme from updating on window resizes
* `buildHook.js`
  * lance's update to get a build hook running to develop with keg-components
* `package.json`
  * updated the version to `2.0.0`
  * Lance updated the `dev` command to use nodemon
* `rollup.config.js`
  * see Lance's PR notes
* `src/dimensions/__tests__/sizeMap.js`
  * updated tests for size keys
* `src/dimensions/sizeMap.js`
  * updated the size key defaults
* `src/hooks/hookFactory.js`
  * see Lance's PR notes
* `src/mocks/deeplyNestedTheme.json`
  * used in unit tests to verify that the bug is fixed that was encountered in keg-components
  * this is a copy of the entire theme object from keg-components, including some size-keys 
* `src/theme/__tests__/buildTheme.js`
  * added a unit test
* `src/theme/__tests__/restructureTheme.js`
   * updated size keys
* `src/theme/buildTheme.js`
 * fixed the bug by ensuring that the theme is merged with the merged theme (which includes the size keys at the root), and not the extracted theme (which has no size keys)
 * updated size keys
* `src/theme/restructureTheme.js`
   * fixed typos

## Testing

* verify the unit tests pass
* run `yarn app` and ensure the app renders
* scroll down to the section called `Screen Dimensions` on the app, and click on the title
* Open the section named `Dynamic theme based on viewport size`
* Scroll down to the rendered component (a div with a background color)
* change the screen size to many different sizes
* verify the background color of the div changes as you change the screen size
